### PR TITLE
Support for ALA's 2019 style

### PR DESCRIPTION
**Pull request by temi**

### CODE REVIEW

File: grails-app/assets/javascripts/head.js [hunk 1]
```diff
@@ -1 +1 @@
-//= require jquery-1.12.4.js
\ No newline at end of file
+//= require jquery-3.4.1.min.js
```
Comment (djtfmartin) [inline grails-app/assets/javascripts/head.js:1]: is jquery 3 essential for this ? 

i propose we can avoid using jquery entirely for this login/logout and therefore avoid the likely versioning issues the apps will hit with this migration to jquery 3.

Interested to know your thoughts @temi Im not sure how extensive the javascript is.

**Review by djtfmartin — CHANGES_REQUESTED** [review]
i propose we can avoid using jquery entirely for this login/logout and therefore avoid the likely versioning issues the apps will hit with this migration to jquery 3.

Interested to know your thoughts @temi Im not sure how extensive the javascript is. Please dont think this is a decision on my part, im just starting a discussion. If we can remove jquery dependency for this plugin, it will make upgrades less painful. If we cant then we'll just go with jquery 3. 

cc @nickdos

**Comment by temi** [comment]
Adding jQuery in the app is an alternative method. But I think it will have its own challenges for the reasons outlined below.

Firstly, existing layout architecture (version 1 & 2) assumes jQuery is provided in the plugin’s layout file. It is either provided in the plugin or retrieved from commonui. Changing this would mean all apps will need to be tested. Therefore will take time.

Secondly, this will require adding jQuery to each app. If I understand correctly, each app will need an intermediary layout file (This file will have the g:applyLayout tag). This will mean updating skin config variable. And, additional effort of making sure jQuery is added in the correct order.

Thirdly, our search bar has autocomplete feature enabled. This is configured in application.js. It assumes a specific jQuery-ui version. I think this will mean more testing to make sure autocomplete works properly with the given jQuery version.

Lastly, a quick browse of vulnerability database showed issues for jQuery version 3.4.0 and lower. Therefore, I decided to go with 3.4.1. I am not a security expert therefore does not know how severe or significant they are. (https://www.cvedetails.com/vulnerability-list.php?vendor_id=6538&product_id=0&version_id=0&page=1&hasexp=0&opdos=0&opec=0&opov=0&opcsrf=0&opgpriv=0&opsqli=0&opxss=0&opdirt=0&opmemc=0&ophttprs=0&opbyp=0&opfileinc=0&opginf=0&cvssscoremin=0&cvssscoremax=0&year=0&cweid=0&order=1&trc=8&sha=cb2a1701a62483883bb26bfed4bac08a56f263d1)

The approach I have taken is to provide the same version of jQuery in both commonui and plugin. This ensures the app will work regardless of app using old or new commonui. Like you mentioned, this might be a bit painful for apps that are yet to be migrated. From my experience so far, it was not that painful since those apps did not use a lot of javascript. Also, I have included the jquery migration plugin to make transition from 2 to 3 painless.

In conclusion, proceeding either way will take time. The proposed change will delay the redesign launch by a few weeks.

**Comment by temi** [comment]
Another important point is changing headerAndFooter config to commonui 2019 should work on any app without upgrading ala-bootstrap3 plugin. By default the login logout buttons are hidden. This works well for apps that does not require users to login. For example, Regions, Dashboard, API, etc. The full list compiled by Nick is [here](https://confluence.csiro.au/display/ALASD/Common+UI+work+for+ALA+sites). Therefore, the list of apps that needs an upgrade are few.

**Comment by djtfmartin** [comment]
Thanks @temi Happy to go ahead with these changes.

**Pull request merged**